src/migrations/tasks/20200126113825_cwu_init.ts

```diff
@@ -0,0 +1,163 @@
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.timestamp('updatedAt').notNullable();
+    table.uuid('updatedBy').references('id').inTable('users').notNullable();
+    table.string('legalName').notNullable();
+    table.string('email').notNullable();
+    table.string('phone');
+    table.string('streetAddress1').notNullable();
+    table.string('streetAddress2');
+    table.string('city').notNullable();
+    table.string('region').notNullable();
+    table.string('mailCode').notNullable();
+    table.string('country').notNullable();
+  });
+  logger.info('Created cwuProponents table.');
+
+  // Add foreign key relationship Proposals --> Proponents
+  await connection.schema.alterTable('cwuProposals', table => {
+    table.uuid('proponent').references('id').inTable('cwuProponents');
+  });
+  logger.info('Foreign key \'proponent\' added to cwuProposals table.');
+
+  // Add constraint to CWUProposal table (proponent is either individual or org - not both)
+  await connection.schema.raw(`
+  ALTER TABLE "cwuProposals"
+  ADD CONSTRAINT "eitherProponentOrOrg" check(
+    ("proponent" IS NOT NULL AND "proponentOrg" IS NULL)
+    OR
+    ("proponent" IS NULL AND "proponentOrg" IS NOT NULL)
+  )
+  `);
+  logger.info('Added constraint \'eitherProponentOrOrg\' to cwuProposals table.');
+
+  // Create CWUProposalAttachment table
+  await connection.schema.createTable('cwuProposalAttachments', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.uuid('proposal').references('id').inTable('cwuProposals').notNullable();
+    table.uuid('file').references('id').inTable('files').notNullable();
+  });
+  logger.info('Created cwuProposalAttachments table.');
+
+  // Create CWUProposalStatus table
+  await connection.schema.createTable('cwuProposalStatuses', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.uuid('proposal').references('id').inTable('cwuProposals').notNullable();
+    table.enu('status', Object.values(ProposalStatus)).notNullable();
+  });
+  logger.info('Created cwuProposalStatuses table.');
+}
+
+export async function down(connection: Knex): Promise<void> {
+  await connection.schema.dropTable('cwuProposalStatuses');
+  await connection.schema.dropTable('cwuProposalAttachments');
+  await connection.schema.raw(`
+    ALTER TABLE "cwuProposals"
+    DROP CONSTRAINT "eitherProponentOrOrg"
+  `);
+  await connection.schema.alterTable('cwuProposals', table => {
+    table.dropColumn('proponent');
+  });
+  await connection.schema.dropTable('cwuProponents');
+  await connection.schema.dropTable('cwuProposals');
```

Comment:
I *think* you can drop the proposals table first and the proponents table second to avoid needing to drop the `proponent` column from the proposals table.

Reply:
Yes, you're right.  I've also removed the `drop constraint` as that's taken care of by dropping the cwuProposals table.